sdk/python/kfp/v2/components/experimental/pipeline_channel.py

```diff
@@ -0,0 +1,321 @@
+        name: str,
+        channel_type: Union[str, Dict],
+        op_name: Optional[str] = None,
+        value: Optional[type_utils.PARAMETER_TYPES] = None,
+    ):
+        """Inits a PipelineArtifactChannel instance.
+
+        Args:
+          name: The name of the pipeline channel.
+          channel_type: The type of the pipeline channel.
+          op_name: Optional; The name of the operation that produces the
+            pipeline channel.
+          value: Optional; The actual value of the pipeline channel.
+
+        Raises:
+          ValueError: If name or op_name contains invalid characters.
+          ValueError: If both op_name and value are set.
+          TypeError: If the channel type is not a parameter type.
+        """
+        if op_name and value:
+            raise ValueError('op_name and value cannot be both set.')
+
+        if not type_utils.is_parameter_type(channel_type):
+            raise TypeError(f'{channel_type} is not a parameter type.')
+
+        self.value = value
+
+        super(PipelineParameterChannel, self).__init__(
+            name=name,
+            channel_type=channel_type,
+            op_name=op_name,
+        )
+
+
+class PipelineArtifactChannel(PipelineChannel):
+    """Represents a pipeline artifact channel.
+
+    Attributes:
+      name: The name of the pipeline channel.
+      channel_type: The type of the pipeline channel.
+      op_name: The name of the operation that produces the pipeline channel.
+        A pipeline artifact channel is always produced by some op.
+      pattern: The serialized string regex pattern this pipeline channel created
+        from.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        channel_type: Union[str, Dict],
+        op_name: str,
+    ):
+        """Inits a PipelineArtifactChannel instance.
+
+        Args:
+          name: The name of the pipeline channel.
+          channel_type: The type of the pipeline channel.
+          op_name: The name of the operation that produces the pipeline channel.
+
+        Raises:
+          ValueError: If name or op_name contains invalid characters.
+          TypeError: If the channel type is not an artifact type.
+        """
+        if type_utils.is_parameter_type(channel_type):
+            raise TypeError(f'{channel_type} is not an artifact type.')
+
+        super(PipelineArtifactChannel, self).__init__(
+            name=name,
+            channel_type=channel_type,
+            op_name=op_name,
+        )
+
+
+def extract_pipeline_channels(payload: str) -> List['PipelineChannel']:
+    """Extracts a list of PipelineChannel instances from the payload string.
+
+    Note: this function removes all duplicate matches.
+
+    Args:
+      payload: A string that may contain serialized PipelineChannels.
+
+    Returns:
+      A list of PipelineChannels found from the payload.
+    """
+    matches = re.findall(
+        r'{{channel:op=([\w\s_-]*);name=([\w\s_-]+);type=([\w\s{}":_-]*);}}',
```

Comment:
I wonder if this should be a private constant above, next to a similar constant representing the f-string template used in `str` method for `PipelineChannel`. That way, when we change one, we'll remember to change the other (and we can add a comment to this effect as well).

Reply:
We won't forget to update both because of the unit tests, but I agree putting them together would be better for readability/discoverability.